lib/intacct/actions/update.rb

```diff
@@ -1,12 +1,22 @@
 module Intacct
   module Actions
     class Update < Base
-      def request(_options)
-        Intacct::XmlRequest.build_xml(client, action) do |xml|
-          xml.function(controlid: "1") do
-            xml.update do
-              xml.send(klass.api_name) do
-                klass.update_xml(xml)
+      def request(options = {})
+        if legacy?
+          Intacct::XmlRequest.build_xml(client, action) do |xml|
+            xml.function(controlid: "1") do
+              xml.send(klass.legacy_update_name, key: klass.public_send(klass.id_attribute)) do
+                update_with_options? ? klass.update_xml(xml, options) : klass.update_xml(xml)
+              end
+            end
+          end
+        else
+          Intacct::XmlRequest.build_xml(client, action) do |xml|
+            xml.function(controlid: "1") do
+              xml.update do
+                xml.send(klass.api_name) do
+                  update_with_options? ? klass.update_xml(xml, options) : klass.update_xml(xml)
+                end
```

Comment:
Can we move these out into methods like `generate_legacy_request` and `generate_request`?